container-storage-setup.sh

```diff
@@ -1109,6 +1110,84 @@ setup_extra_volume() {
   fi
 }
 
+# This is used only in compatibility mode. We are still using systemd
+# mount unit only for compatibility mode. Reason being that upon service
+# restart, we run into races and device might not yet be up when we try
+# to mount it. And we don't know if this is first time start and we need
+# to create volume or this is restart and we need to wait for device. So
+# continue to use systemd mount unit for compatibility mode.
+setup_systemd_mount_unit_compat() {
+  local filename
+  local vg=$1
+  local lv_name=$2
+  local mount_dir=$3
+  local unit_file_path
+
+  # filename must match the path ${mount_dir}.
+  # e.g if ${mount_dir} is /var/lib/containers
+  # then filename will be var-lib-containers.mount
+  filename=$(systemd_escaped_filename ${mount_dir})
+  unit_file_path="/etc/systemd/system/$filename"
```

Comment:
So I will have to write a generator of my own? How it is different than what I am doing now? I am also generating a unit file. Just that outside the systemd mechanism to call generators.

Also I am assuming that I need to drop a generator binary somewhere (instead of script) and we don't compile anything in this package.

So for this use case, it feels like that switching to generator mechanism might be work and less gain.